src/parser.ts

```diff
@@ -939,7 +939,8 @@ export class Parser extends DiagnosticEmitter {
 
     // at '<': TypeParameter (',' TypeParameter)* '>'
 
-    var typeParameters = new Array<TypeParameterNode>();
+    // var typeParameters = new Array<TypeParameterNode>();
+    var typeParameters: TypeParameterNode[] | null = null;
```

Comment:
Now I'm wondering: If we are already at `<`, the usual case here is that we'll end up with an array and not `null` since `<>` is invalid. If we had the `Array.create<TypeParameterNode>(1)` helper I mentioned in the runtime PR (the one that would guarantee a backing buffer for at least one element but with length=0), that might make the most sense here as it avoids the (most of the time) redundant null check below. Wdyt?

Reply:
I think it could be even better. Reserve not 1 element but for example 8 when array is created. And this element btw could store outside buffer exactly inside Array class. Similar approach have [SmallVector in binaryen](https://github.com/WebAssembly/binaryen/blob/29a18ccb610defbab7efb505f3a0531c0e55403f/src/support/small_vector.h#L32) and [LLVM](http://llvm.org/doxygen/classllvm_1_1SmallVector.html)